lib/MadNLPGPU/src/interface.jl

```diff
@@ -2,10 +2,29 @@
 function CuInteriorPointSolver(nlp::AbstractNLPModel{T};
     option_dict::Dict{Symbol,Any}=Dict{Symbol,Any}(), kwargs...
 ) where T
-    opt = MadNLP.Options(linear_solver=LapackGPUSolver)
-    MadNLP.set_options!(opt,option_dict,kwargs)
+    # Initiate interior-point options
+    opt = MadNLP.IPMOptions(linear_solver=LapackGPUSolver)
+    linear_solver_options = set_options!(opt, kwargs)
     MadNLP.check_option_sanity(opt)
 
+    # Initiate linear-solver options
```

Comment:
line 10-26 seems to be redundant to line 104-120 of IPM.jl. Can we create an internal function for setting options?